setup/plugins/init/Drupal8.civi-setup.php

```diff
@@ -43,7 +43,7 @@
     $ssl_params = \Civi\Setup\DrupalUtil::guessSslParams($connectionOptions);
     // @todo Does Drupal support unixsocket in config? Set 'server' => 'unix(/path/to/socket.sock)'
     $model->db = $model->cmsDb = array(
-      'server' => \Civi\Setup\DbUtil::encodeHostPort($connectionOptions['host'], $connectionOptions['port'] ?: NULL),
+      'server' => \Civi\Setup\DbUtil::encodeHostPort($connectionOptions['host'], empty($connectionOptions['port']) ? NULL : $connectionOptions['port']),
```

Comment:
```suggestion
      'server' => \Civi\Setup\DbUtil::encodeHostPort($connectionOptions['host'], $connectionOptions['port'] ?? NULL),
```

Reply:
I thought about that but there are times it comes thru as the empty string, and the intent seemed to be to convert that to null. It seems like php could use yet another operator, like `$a ¿ $b`, meaning "if $a is empty use $b otherwise $a".

Reply:
I suppose this might work, but I'm not sure it's more readable? `($connectionOptions['port'] ?? NULL) ?: NULL`

Reply:
No I don't think we need to convert empty strings to null. If you look at the `encodeHostPort` function, it treats both exactly the same.

Reply:
Ok